packages/theme-language-server-common/src/TypeSystem.ts

```diff
@@ -122,8 +127,24 @@ export class TypeSystem {
         ...(objectMap.settings ?? {}),
         properties: themeSettingProperties,
       },
+      ...customMetafieldTypeEntries(objectMap['metafield']),
+      ...metafieldDefinitionsObjectMap,
     };
 
+    // For each metafield definition fetched, we need to override the property from the vanilla
+    // `metafield` type to `${category}_metafield`.
+    for (let category of FETCHED_METAFIELD_CATEGORIES) {
+      if (!result[category]) continue;
+
+      let metafieldsProperty = result[category].properties?.find(
+        (prop) => prop.name === 'metafields',
+      );
+
+      if (!metafieldsProperty) continue;
+
+      metafieldsProperty.return_type = [{ type: `${category}_metafields`, name: '' }];
```

Comment:
Hmm. Thinking about this, are we sure we're not mutating the original object here? Should we clone/replace metafieldsProperty before doing this mutation? 

Remember that `...objectMap` gives you a _shallow_ clone of the object map. Looks like we're mutating the original object map here and this is no bueno. Means that if you'd get metafield data from one theme in another.

Reply:
We talked about it in person - since this isn't unique to a theme, we should be fine with this change happening across themes within a workspace. Added a warning message in a comment above the block instead.